packages/plexus/src/LayoutManager/Coordinator.tsx

```diff
@@ -176,31 +181,37 @@ export default class Coordinator {
   };
 
   _handleVizWorkerMessage = (event: MessageEvent) => {
-    const worker: LayoutWorker = ((event.target: any): LayoutWorker);
-    const workerMessage = ((event.data: any): WorkerMessage);
-    const { type, meta } = workerMessage;
+    const worker: LayoutWorker = event.target as LayoutWorker;
+    const workerMessage = event.data as TWorkerOutputMessage | TWorkerErrorMessage;
+    const { type } = workerMessage;
     this._makeWorkerIdle(worker);
-    const isStale = !this.currentLayout || meta.layoutId !== this.currentLayout.id;
-    if (type === 'error') {
-      console.error(`${isStale ? '(stale) ' : ' '}Error from viz.worker`, event);
+    const isStale =
+      !this.currentLayout ||
+      Boolean(workerMessage.meta && workerMessage.meta.layoutId !== this.currentLayout.id);
+    if (type === EWorkerErrorType.Error) {
+      console.error(`${isStale ? '(stale) ' : ''}Error from viz.worker`, event);
       return;
     }
     if (isStale) {
       return;
     }
-    if (type === 'layout-error') {
+    if (type === EWorkerErrorType.LayoutError) {
```

Comment:
this if statement, the following and the proceeding `type ===` are also comparing values to types. Maybe that's just a standard for `Enum`s that I'm not used to but it seems as though they shouldn't be used in that way.

Reply:
These are comparing values to values. The type is `EWorkerErrorType` and `EWorkerErrorType.LayoutError` is a value with that type. So, the code above is comparing a value with a value.